src/hotspot/share/jvmci/jvmci.cpp

```diff
@@ -130,11 +130,6 @@ void JVMCI::ensure_box_caches_initialized(TRAPS) {
   if (_box_caches_initialized) {
     return;
   }
-  MutexLocker locker(JVMCI_lock);
```

Comment:
This lock is required in the case where this is not called from `AOTLoader::initialize_box_caches` to ensure only one JVMCI compiler thread does the lazy initialization.
I can fix this if you'd like me to put up a separate PR.

Reply:
OK. Please feel free to do so.
Thanks.

Reply:
I've opened https://github.com/openjdk/jdk/pull/1730.